frontend/src/component/SummaryCard.tsx

```diff
@@ -1,20 +1,59 @@
 import {Summary} from "../type/Summary.tsx";
+import ButtonWithIcon from "./ButtonWithIcon.tsx";
+import html2canvas from "html2canvas";
+import jsPDF, {ImageProperties} from "jspdf";
+import {useRef} from "react";
 
 export type SummaryCardProps = {
-    summary: Summary | null
+    summary: Summary | null,
+    onClick: () => void,
 }
 
-export default function SummaryCard({summary}: SummaryCardProps){
+export default function SummaryCard({summary, onClick}: SummaryCardProps){
+
+    const printRef = useRef(null)
 
     if (summary?.text === undefined) {
-        return "Keine Zusammenfassung gefunden!";
+        return console.error("Keine Zusammenfassung gefunden!");
+    }
+
+    // String in Array to separate the different bullet points
+    const summaryPoints: string[] = summary?.text.split("\n");
+
+    // Button handler to edit summary
+    const handleEditButtonClick = () => {
+        onClick();
     }
 
-    const summaryPoints = summary?.text.split("\n"); // String in Array
+    // Button handler to generate a PDF
+    const handlePDFButtonClick = async () => {
+        const element = printRef.current;
+        if (!element) {
+            return;
+        }
+
+        const canvas: HTMLCanvasElement = await html2canvas(element);
+        const data: string = canvas.toDataURL("image/png");
```

Comment:
You probably have a good reason for handling the pdf as image, but have you considered making the text selectable? (I have no idea how though)